src/server/routes.pl

```diff
@@ -2488,6 +2492,36 @@
         )
     ).
 
+/*
+ * patch_handler(Mode, Request, System, Auth) is det.
+ *
+ * Reset a branch to a new commit.
+ */
+patch_handler(post, Path, Request, System_DB, Auth) :-
+    do_or_die(
+        (   get_payload(JSON, Request),
+            _{ patch : Patch } :< JSON),
+        error(bad_api_document(JSON, [patch]), _)),
+    % We should take options about final state here.
+    api_report_errors(
+        patch,
+        Request,
+        (   (   memberchk(search(Search), Request)
+            ->  true
+            ;   Search = []),
+            param_value_search_or_json_required(Search, JSON, author, text, Author),
```

Comment:
There is a weird inconsistency here between endpoints.
Some endpoints seem to expect commit info to be in a special commit_info subdocument in the payload. But this (and I presume other places where this was copied from) would expect author and message to be toplevel.

Reply:
Let's fix that where the inconsistency arises.